### ISSUE TITLE
Kani gives wrong result if two harnesses have the same name

### ISSUE THREAD

**Issue by celinval** [model-checking/kani#1974]
Here are the steps to reproduce them:

1. Create two crates, and add a harness to each of them with the same name, e.g.:

```
cargo init --lib lib1
cargo init --lib lib2
```

Where `lib1/src/lib.rs` include:

```rust
#[kani::proof]
fn check_lib() {
    assert!(1 == 1, "This should succeed");
}

```

And `lib2/src/lib.rs` include:

```rust
#[kani::proof]
fn check_lib() {
    assert_eq!(1 + 1, 1, "This should fail");
}
```

2. Create a workspace that includes both crates:

```toml
[workspace]
members = [
  "lib1",
  "lib2",
]
```

3. Run `cargo kani` in the workspace.

with Kani version: 0.16.0

I expected to see this happen: Kani should report that one harness failed and one succeeded.

Instead, this happened: Kani claims that both harnesses succeed.

```
$ cargo kani --output-format=terse
   Compiling lib1 v0.1.0 (/tmp/ws-name-conflict/lib1)
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
   Compiling lib2 v0.1.0 (/tmp/ws-name-conflict/lib2)
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
Checking harness check_lib...

VERIFICATION RESULT:
 ** 0 of 1 failed

VERIFICATION:- SUCCESSFUL
Verification Time: 0.01958095s

Checking harness check_lib...

VERIFICATION RESULT:
 ** 0 of 1 failed

VERIFICATION:- SUCCESSFUL
Verification Time: 0.019129628s

Complete - 2 successfully verified harnesses, 0 failures, 2 total.
```
